protocol/messenger_profile_showcase.go

```diff
@@ -71,69 +86,102 @@ func toProfileShowcaseCollectibleProto(preferences []*ProfileShowcaseCollectible
 		})
 	}
-	return collectibles
+	return entries
 }
 
-func toProfileShowcaseVerifiedTokensProto(preferences []*ProfileShowcaseVerifiedTokenPreference, visibility ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseVerifiedToken {
-	tokens := []*protobuf.ProfileShowcaseVerifiedToken{}
+func (m *Messenger) toProfileShowcaseVerifiedTokensProto(preferences []*identity.ProfileShowcaseVerifiedTokenPreference, visibility identity.ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseVerifiedToken {
+	entries := []*protobuf.ProfileShowcaseVerifiedToken{}
 	for _, preference := range preferences {
 		if preference.ShowcaseVisibility != visibility {
 			continue
 		}
 
-		tokens = append(tokens, &protobuf.ProfileShowcaseVerifiedToken{
+		entries = append(entries, &protobuf.ProfileShowcaseVerifiedToken{
 			Symbol: preference.Symbol,
 			Order:  uint32(preference.Order),
 		})
 	}
-	return tokens
+	return entries
 }
 
-func toProfileShowcaseUnverifiedTokensProto(preferences []*ProfileShowcaseUnverifiedTokenPreference, visibility ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseUnverifiedToken {
-	tokens := []*protobuf.ProfileShowcaseUnverifiedToken{}
+func (m *Messenger) toProfileShowcaseUnverifiedTokensProto(preferences []*identity.ProfileShowcaseUnverifiedTokenPreference, visibility identity.ProfileShowcaseVisibility) []*protobuf.ProfileShowcaseUnverifiedToken {
+	entries := []*protobuf.ProfileShowcaseUnverifiedToken{}
 	for _, preference := range preferences {
 		if preference.ShowcaseVisibility != visibility {
 			continue
 		}
 
-		tokens = append(tokens, &protobuf.ProfileShowcaseUnverifiedToken{
+		entries = append(entries, &protobuf.ProfileShowcaseUnverifiedToken{
 			ContractAddress: preference.ContractAddress,
 			ChainId:         preference.ChainID,
 			Order:           uint32(preference.Order),
 		})
 	}
-	return tokens
+	return entries
 }
 
-func fromProfileShowcaseCommunityProto(messages []*protobuf.ProfileShowcaseCommunity) []*ProfileShowcaseCommunity {
-	communities := []*ProfileShowcaseCommunity{}
-	for _, entry := range messages {
-		communities = append(communities, &ProfileShowcaseCommunity{
-			CommunityID: entry.CommunityId,
-			Order:       int(entry.Order),
+func (m *Messenger) fromProfileShowcaseCommunityProto(senderPubKey *ecdsa.PublicKey, messages []*protobuf.ProfileShowcaseCommunity) []*identity.ProfileShowcaseCommunity {
+	entries := []*identity.ProfileShowcaseCommunity{}
+	for _, message := range messages {
+		entry := &identity.ProfileShowcaseCommunity{
+			CommunityID: message.CommunityId,
+			Order:       int(message.Order),
+		}
+
+		community, err := m.FetchCommunity(&FetchCommunityRequest{
+			CommunityKey:    message.CommunityId,
+			Shard:           nil,
+			TryDatabase:     true,
+			WaitForResponse: true,
 		})
+		if err != nil {
+			m.logger.Warn("failed to fetch community for profile entry ", zap.Error(err))
+		}
+
+		if community != nil && community.Encrypted() {
```

Comment:
If we don't know the community, do we just ignore it or should we mark is as unknown?

Reply:
fixed, entry will be `ProfileShowcaseMembershipStatusUnproven` by default